src/net.cpp

```diff
@@ -1497,34 +1493,29 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
 
 void CConnman::SocketHandler()
 {
-    std::set<SOCKET> recv_set, send_set, error_set;
-    SocketEvents(recv_set, send_set, error_set);
-
-    if (interruptNet) return;
+    std::set<SOCKET> recv_set;
+    std::set<SOCKET> send_set;
+    std::set<SOCKET> error_set;
```

Comment:
4097d14 unneeded style change, would leave as-is (and as currently exists in both of the `SocketEvents()` functions)
```diff
-    std::set<SOCKET> recv_set;
-    std::set<SOCKET> send_set;
-    std::set<SOCKET> error_set;
-
+    std::set<SOCKET> recv_set, send_set, error_set;
```

Reply:
I think it is good to always declare one variable per line.

Reply:
Consider:
```cpp
struct SocketSets {
    std::set<SOCKET> recv;
    std::set<SOCKET> send;
    std::set<SOCKET> error;
}
```

Reply:
@LarryRuane, something like this is coming as a subsequent change from https://github.com/bitcoin/bitcoin/pull/21878 (see commit `net: introduce Sock::WaitMany()` and the structures `WaitEvents` and `WaitData` introduced in that commit).

The current combination of sets stores the sockets in "one pile of sockets that are ready for read" and "one pile of sockets that are ready for write" (notice that a given socket may be in both). It is more convenient to have just one pile of sockets each one with attached ready-for-read and ready-for-write flags.

Reply:
I like the idea of abstracting this but it seems out of scope of this PR.

> It is more convenient to have just one pile of sockets each one with attached ready-for-read and ready-for-write flags.

Eventually it depends on the selection mechanism used isn't it? When it's abstracted, might as well store it in a format ready to feed to say, `poll()` or `select()` whatever is used.

Reply:
Yes, those 3 sets are dragged around as parameters to functions and as local variables. It is an obvious improvement to pack them together somehow. Not the purpose of this PR. Once this is PR is merged I will chop off another piece from #21878 into a smaller/manageable PR that will contain the commit which introduces such a packing `net: introduce Sock::WaitMany()`. We can discuss it there, I don't have a strong opinion on how they are packed exactly. Or if this PR is stuck we can move the discussion to #21878 where `WaitData` is introduced.

Reply:
> Once this is PR is merged I will chop off another piece from #21878 into a smaller/manageable PR that will contain the commit which introduces such a packing `net: introduce Sock::WaitMany()`

There are too many conflicts if I try to cherry-pick the `WaitMany()` stuff on top of bare `master` - it touches the same areas of code as the preceding commits from #21878. I chopped off those preceding commits in separate PRs instead: https://github.com/bitcoin/bitcoin/pull/23601 and https://github.com/bitcoin/bitcoin/pull/23604.

```
WaitMany | net: use Sock::WaitMany() instead of CConnman::SocketEvents()
stuff    | net: introduce Sock::WaitMany()
         | net: also wait for exceptional events in Sock::Wait()

#23601   | net: don't check if the listening socket is valid

         | scripted-diff: rename CNode::cs_hSocket to CNode::m_sock_mutex
#23604   | net: use Sock in CNode
         | fuzz: move FuzzedSock earlier in src/test/fuzz/util.h

         | net: change CreateNodeFromAcceptedSocket() to take Sock
#21879   | net: use Sock in CConnman::ListenSocket
         | net: add new method Sock::Accept() that wraps accept()
```